src/route/dataset-route.ts

```diff
@@ -39,72 +48,81 @@ function checkDatasetID(datasetID: string, res: Response): boolean {
+    return {
+        success: false,
+        dataset_id: datasetID,
+        errors: [
+            {
+                field,
+                message: [
+                    {
+                        lang: ENGLISH,
+                        message: t(translationString, { lng: ENGLISH })
+                    },
+                    {
+                        lang: WELSH,
+                        message: t(translationString, { lng: WELSH })
                     }
+                ],
+                tag: {
+                    name: translationString,
+                    params: {}
                 }
-            ]
-        };
+            }
+        ]
+    };
+}
+
+
+apiRoute.post('/', upload.single('csv'), async (req: Request, res: Response) => {
+    if (!req.file) {
         res.status(400);
-        res.json(err);
+        res.json(errorDtoGenerator('csv', 'errors.no_csv_data'));
         return;
     }
     const lang: string = req.body?.language || req.i18n.language;
     const title: string = req.body?.title;
     if (!title) {
-        const err: ViewErrDTO = {
-            success: false,
-            dataset_id: undefined,
-            errors: [
-                {
-                    field: 'title',
-                    message: [
-                        {
-                            lang: ENGLISH,
-                            message: t('errors.no_title', { lng: ENGLISH })
-                        },
-                        {
-                            lang: WELSH,
-                            message: t('errors.no_title', { lng: WELSH })
-                        }
-                    ],
-                    tag: {
-                        name: 'errors.no_title',
-                        params: {}
-                    }
-                }
-            ]
-        };
         res.status(400);
-        res.json(err);
+        res.json(errorDtoGenerator('title', 'errors.no_title'));
         return;
     }
-    const dataset = Dataset.createDataset(title, 'BetaUser');
-    const saved_dataset_record = await dataset.save();
-    saved_dataset_record.addTitleByString(title, lang);
-    const uploadDTO = await uploadCSVToBlobStorage(req.file?.buffer, saved_dataset_record);
-    if (!uploadDTO.success) {
-        res.status(400);
+    let importRecord: Import;
+    try {
+        importRecord = await uploadCSVToBlobStorage(req.file?.stream, req.file?.mimetype);
+    } catch (err) {
+        logger.error(`An error occured trying to upload the file with the following error: ${e}`);
+        res.status(500);
+        res.json({ message: 'Error uploading file' });
+        return;
     }
+
+    // Everything looks good so far, let's create the dataset and revision records
+    const dataset = new Dataset();
+    dataset.creation_date = new Date();
+    // TODO change how we handle authentication to get the user on the Backend
+    dataset.created_by = 'Test User';
+    const saved_dataset_record = await dataset.save();
```

Comment:
We don't need to await the DB transaction here as the object should have everything it needs for us to link to it in other objects later.

Reply:
We await the save so we can get the ID from the saved object.

Reply:
It's a UUID: we can (and should) assign it ourselves. That's the point of UUIDs.

Reply:
That said I might just batch up object creation and save the overall dataset and see what happens in test.